packages/editor/src/components/toolbar/toolbar-format-select/index.ts

```diff
@@ -0,0 +1,117 @@
+import type { Editor } from '@tiptap/core';
+import { consume } from '@lit/context';
+import { html, LitElement } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { map } from 'lit/directives/map.js';
+import { tiptapContext } from '@/context/tiptapContext.ts';
+import buttonStyles from './../toolbar-button/styles.ts';
+
+export interface SelectOption {
+  active: boolean;
+  label: string;
+  value: string;
+}
+
+@customElement('clippy-format-select')
+export class FormatSelect extends LitElement {
+  @property({ type: Boolean }) disabled = false;
+  @property({ type: Boolean }) readOnly = false;
+  @property({ type: Function }) onSelect = (value: string) => value;
+  @consume({ context: tiptapContext, subscribe: true })
+  @property({ attribute: false })
+  public editor?: Editor;
+  static override readonly styles = [buttonStyles];
+
+  #isFormatActive(name: string, attributes?: Record<'level', number>): boolean {
+    return this.editor?.isActive(name, attributes) ?? false;
+  }
+
+  readonly #handleTextFormatChange = (event: Event) => {
+    const { value } = event.target as HTMLSelectElement;
+
+    if (!value || !this.editor) return;
+
+    const chain = this.editor.chain().focus();
+
+    const formatCommands: Record<string, () => typeof chain> = {
+      h1: () => chain.toggleHeading({ level: 1 }),
+      h2: () => chain.toggleHeading({ level: 2 }),
+      h3: () => chain.toggleHeading({ level: 3 }),
+      h4: () => chain.toggleHeading({ level: 4 }),
+      h5: () => chain.toggleHeading({ level: 5 }),
+      h6: () => chain.toggleHeading({ level: 6 }),
+      paragraph: () => chain.setParagraph(),
+    };
+    formatCommands[value]().run();
+  };
+
+  readonly #onUpdate = () => this.requestUpdate();
+
+  override connectedCallback() {
+    super.connectedCallback();
+    this.editor?.on('transaction', this.#onUpdate);
+  }
+
+  override disconnectedCallback() {
+    this.editor?.off('transaction', this.#onUpdate);
+    super.disconnectedCallback();
+  }
+
+  @state()
+  private get options(): SelectOption[] {
```

Comment:
```suggestion
  get #options(): SelectOption[] {
```

Geloof dat die # ook gewoon bij n getter mag, maar let op dat je dan ook`this.#options` moet doen

Reply:
Onder een `@state()` krijg ik TS1206: Decorators are not valid here.